spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheAutoConfiguration.java

```diff
@@ -134,4 +142,45 @@ public void afterPropertiesSet() {
 
 	}
 
+	/**
+	 * Exception thrown when {@link org.springframework.cache.CacheManager} implementation
+	 * are not specified.
+	 */
+	static class NoCacheManagerException extends RuntimeException {
+
+		private final CacheProperties cacheProperties;
+
+		NoCacheManagerException(CacheProperties cacheProperties) {
+			super("No cache manager could be auto-configured, check your configuration"
+					+ " (caching type is '" + cacheProperties.getType() + "')");
+			this.cacheProperties = cacheProperties;
+		}
+
+		CacheProperties getCacheProperties() {
+			return this.cacheProperties;
+		}
+
+	}
+
+	/**
+	 * Exception thrown when multiple {@link org.springframework.cache.CacheManager}
+	 * implementations are available with no way to know which implementation should be
+	 * used.
+	 */
+	static class NonUniqueCacheManagerException extends RuntimeException {
```

Comment:
And this one could be `NoUniqueCacheManagerException` extending from the other exception (again to mirror `NoUniqueBeanDefinitionException`)